contracts/automation/andromeda-task-balancer/src/contract.rs

```diff
@@ -0,0 +1,428 @@
+        }
+        _ => Ok(Response::default()),
+    }
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    let contract = ADOContract::default();
+    match msg {
+        ExecuteMsg::AndrReceive(msg) => contract.execute(deps, env, info, msg, execute),
+        ExecuteMsg::Add { process } => add_process(deps, env, info, process),
+        ExecuteMsg::UpdateAdmin { new_admin } => update_admin(deps, info, new_admin),
+        ExecuteMsg::Remove { process } => remove_process(deps, env, info, process),
+    }
+}
+
+fn remove_process(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    process: String,
+) -> Result<Response, ContractError> {
+    // Add permission for removal of processes
+    // Sender should be part of an already existing process, and can't request the removal of another process
+
+    // Identify which storage contract a certain process belongs to
+    // Get storage address that holds the process
+    let mut num = 0;
+    let contract_addr = STORAGE_CONTRACTS.load(deps.storage)?;
+
+    let storage_address = loop {
+        let has_process = deps.querier.query(&QueryRequest::Wasm(WasmQuery::Smart {
+            contract_addr: contract_addr[num as usize].clone(),
+            msg: to_binary(&StorageQueryMsg::HasProcess {
+                process: deps.api.addr_validate(&process)?,
+            })?,
+        }))?;
+        if has_process {
+            break &contract_addr[num as usize];
+        }
+        num += 1;
+
+        if num as usize >= (contract_addr.len()) {
+            return Err(ContractError::ProcessNotFound {});
+        }
+    };
+    // Mark that storage contract by adding it to the UP_NEXT vector, which allows us to fill that empty space in the future
+    let up_next = UP_NEXT.may_load(deps.storage)?;
+
+    if let Some(mut up_next) = up_next {
+        up_next.push(storage_address.clone());
+        UP_NEXT.save(deps.storage, &up_next)?;
+    } else {
+        let up_next = vec![storage_address.clone()];
+        UP_NEXT.save(deps.storage, &up_next)?;
+    }
+
+    Ok(Response::new()
+        .add_submessage(SubMsg::new(CosmosMsg::Wasm(WasmMsg::Execute {
+            contract_addr: storage_address.clone(),
+            msg: to_binary(&StorageExecuteMsg::Remove {
+                process: process.to_string(),
+            })?,
+            funds: vec![],
+        })))
+        .add_attribute("action", "removed_process")
+        .add_attribute("process", process)
+        .add_attribute("up_next", storage_address))
+}
+
+fn add_process(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    process: String,
+) -> Result<Response, ContractError> {
+    // Not anyone should be allowed to add tasks to tree
+    let state = STATE.load(deps.storage)?;
+    ensure!(info.sender == state.admin, ContractError::Unauthorized {});
```

Comment:
Why do we need an admin specifically? Can we not use operators or owner from ADO functionality?